ui/apps/platform/src/Containers/Clusters/DelegateScanning/Components/DelegatedRegistriesTable.tsx

```diff
@@ -0,0 +1,300 @@
+                evt.clientX < ulRect.x + ulRect.width &&
+                evt.clientY > ulRect.y &&
+                evt.clientY < ulRect.y + ulRect.height
+            );
+        }
+        return false;
+    }
+
+    const onDrop: TrProps['onDrop'] = (evt) => {
+        if (isValidDrop(evt)) {
+            setItemOrder(tempItemOrder);
+
+            // the rest of this block was added to the PF drag and drop paradigm,
+            // in order to keep the form data in sync with PF's visual drop order
+            const newRegistries: DelegatedRegistry[] = tempItemOrder.map((tempItem) => {
+                const newIndex = registries.findIndex((reg) => reg.uuid === tempItem) || 0;
+                return registries[newIndex];
+            });
+
+            updateRegistriesOrder(newRegistries);
+        } else {
+            onDragCancel();
+        }
+    };
+
+    function onDragOver(evt): TbodyProps['onDragOver'] {
+        evt.preventDefault();
+
+        const curListItem = (evt.target as HTMLTableSectionElement).closest('tr');
+        if (
+            !curListItem ||
+            !bodyRef?.current?.contains(curListItem) ||
+            curListItem.id === draggedItemId
+        ) {
+            return undefined;
+        }
+        const dragId = curListItem.id;
+        const newDraggingToItemIndex = Array.from(bodyRef.current.children).findIndex(
+            (item) => item.id === dragId
+        );
+        if (newDraggingToItemIndex !== draggingToItemIndex) {
+            const tempItemOrder = moveItem(
+                [...itemOrder],
+                draggedItemId || '',
+                newDraggingToItemIndex
+            );
+            move(tempItemOrder);
+            setDraggingToItemIndex(newDraggingToItemIndex);
+            setTempItemOrder(tempItemOrder);
+        }
+
+        return undefined;
+    }
+
+    const onDragEnd: TrProps['onDragEnd'] = (evt) => {
+        const target = evt.target as HTMLTableRowElement;
+        target.classList.remove(styles.modifiers.ghostRow);
+        target.setAttribute('aria-pressed', 'false');
+        setDraggedItemId(null);
+        setDraggingToItemIndex(null);
+        setIsDragging(false);
+    };
+    // End PatternFly template for drag and drop
+
+    return (
+        <TableComposable
+            aria-label="Delegated registry exceptions table"
+            className={(isDragging && styles.modifiers.dragOver) || ''}
+        >
+            <Thead>
+                <Tr>
+                    <Th>Order</Th>
+                    <Th width={40}>Source registry</Th>
+                    <Th width={40}>Destination cluster (CLI/API only)</Th>
+                    <Td isActionCell />
+                </Tr>
+            </Thead>
+            <Tbody
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                // @ts-ignore
+                ref={bodyRef}
+                onDragOver={onDragOver}
+                onDrop={onDragOver}
```

Comment:
i found this odd when looking at it -- why would we need to pass through `onDragOver` twice? It feels odd to me that we can't pass through `onDrop` here to follow the semantics of all the rest of the drag/drop functions.

Reply:
It is the pattern used in the PF4 docs,
https://v4-archive.patternfly.org/v4/components/table#composable-draggable-row-table

and is still that way for this Table variant in the PF5 docs,
https://www.patternfly.org/components/table#draggable-row-table

The `onDrop` handler is passed as a prop to the Tr element below.